lib/model/autocomplete.dart

```diff
@@ -111,34 +122,57 @@ class AutocompleteIntent {
 ///
 /// On reassemble, call [reassemble].
 class AutocompleteViewManager {
-  final Set<MentionAutocompleteView> _mentionAutocompleteViews = {};
+  final Map<Type, Set<AutocompleteView>> _views = {};
```

Comment:
The motivation behind using this collection is to provide a generic way for registering and unregistering the vm without having to repeat the some code blocks and instead having that logic implemented in the parent `AutocompleteView` making it a little bit cleaner and less complicated for subclasses implementing it.

Having in mind that we don't expect to have a big number of kinds of `AutocompleteView` this is completely optional.

Reply:
Yeah, I think this will be simplest if it's a separate variable for each type of view, rather than a `Map<Type, …>`.

It means a few similar-looking registerFoo / unregisterFoo methods. But each of them will be quite short, so there's not much gained by deduplicating them.